texastribune/Elements
# Add touch listeners to dialog buttons

#### What's this PR do?

- There are three potential buttons in a dialog element: expand, delete, and confirm. The expand button will expand the dialog to fit the entire screen. The delete button will hide the dialog. The confirm button will confirm whatever is in the dialog (in the case of the files app, the confirmation button will add an image to the directory). These buttons previously had `mousedown` and `click` event handlers, but no touch event handlers for mobile devices. This PR adds touch event handlers for all three of those buttons.
- Updates README and `package.json` with some useful commands
- Bumps `package-lock.json` to use lock version 2.

#### Related PRs

texastribune: https://github.com/texastribune/texastribune/pull/4677
tt-docker-base: https://github.com/texastribune/tt-docker-base/pull/599
files: https://github.com/texastribune/files/pull/508

#### Why are we doing this? How does it help us?

Photo staffers may want to upload photos on their phones to the CMS. This PR allows them to do that. And it allows people to hide/expand a dialog as they would be able to do on desktop.

#### How should this be manually tested?

_In the docs_

1. Go to <https://texastribune.github.io/Elements/> on your desktop
2. Click on Examples
3. Click on the dialog tab
4. Note that you can click the expand and exit buttons in the top right of each dialog. Only clicking outside of the dialog boxes will make them disappear
5. Go to the same page on your mobile device and note that those two buttons do not work
6. Clone this repository and install dependencies with `npm install`
7. Generate new docs by running `npm run doc`
8. Spin up a local server at the `docs` directory
9. Run `ifconfig` to get your local network IP address
10. Go to the network address on your phone (with the proper port)
11. Go to the dialog tab and note that the buttons work now
12. Note that the expected behavior still works on desktop

#### How should this change be communicated to end users?

Send a note in #team-engineering

#### Are there any smells or added technical debt to note?

There are some strange dialog behaviors in this documentation in particular, but they don't seem to happen in the files app or texastribune app. Strange behavior includes:
- On mobile, dragging a dialog box will move it and expand it at the same time. This happens regardless of this PR.
- On mobile, expanding a dialog box will sometimes make it move off screen.

#### What are the relevant tickets?

<https://airtable.com/appyo1zuQd8f4hBVx/tbloNZu8GkM52NKFR/viwPxWENGdckjxRrP/recvM0Tfl61zz19kp?blocks=hide>

#### Have you done the following, if applicable:
***(optional: add explanation between parentheses)***

* [x] Tested manually on mobile? *(See manual testing)*

#### TODOs / next steps:

* [x] Merge into master
* [ ] Update docs by running `npm run doc`
